File: Preparation_Python/2025/ValueOf_nPr.py
def fact(n):
    fact = 1
    for i in range(1,n+1):
        fact *= i
    return fact
def nPr(n,r):
    return fact(n) / fact(n-r)

n = int(input())
r = int(input())
print(nPr(n,r))

# nPr = n! / (n - r)! 

# Examples:

# Input: n = 5
#            r = 2
# Output: 20
# Explanation: 5P2 = 5! / (5 - 2)!  = 20

# Input: n = 6
#            r = 3
# Output: 120
# Explanation: 6P3 = 6! / (6 - 3)!  = 120

File: Medium_problems/ValueOf_nPr.java
//nPr = n! / (n - r)! 
//
//Examples:
//
//Input: n = 5
//           r = 2
//Output: 20
//Explanation: 5P2 = 5! / (5 - 2)!  = 20
//
//Input: n = 6
//           r = 3
//Output: 120
//Explanation: 6P3 = 6! / (6 - 3)!  = 120

package Medium_problems;

import java.util.Scanner;

public class ValueOf_nPr {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int r = sc.nextInt();
		
        System.out.println(n + "P" + r + " = " + nPr(n, r));
	}
	
	public static int factorial(int n) {
		int fact = 1;
		for(int i=2; i<=n; i++) {
			fact *= i;
		}
		return fact;
	}
	
	public static long nPr(int n,int r) {
		return factorial(n) / factorial(n-r);
	}

}
